src/server.py

```diff
@@ -33,14 +33,31 @@ def play(self):
     def stop(self):
         self.pipeline.send_event(Gst.Event.new_eos())
 
+    @property
+    def volume(self):
+        if "volume" in self.metadata:
+            return self.metadata["volume"]
+        return None
+
     @property
     def sound_location(self):
         return self.metadata["sound-file"]
 
     def _build_pipeline(self):
-        spipeline = ("filesrc location=\"%s\" ! decodebin ! autoaudiosink" %
-                     self.sound_location)
-        return Gst.parse_launch(spipeline)
+        if self.volume is None:
+            spipeline = "filesrc name=src ! decodebin ! autoaudiosink"
+        else:
+            spipeline = ("filesrc name=src ! decodebin ! volume name=volume "
+                         "! autoaudiosink")
+        pipeline = Gst.parse_launch(spipeline)
+
+        source = pipeline.get_by_name("src")
+        source.set_property("location", self.sound_location)
+        if self.volume is not None:
+            volume = pipeline.get_by_name("volume")
+            volume.set_property("volume", self.volume)
```

Comment:
You can put the property directly in the pipeline text description: `'volume volume={}'.format(volume)`